dali/pipeline/util/thread_pool.cc

```diff
@@ -90,15 +90,16 @@ void ThreadPool::WaitForWork(bool checkForErrors) {
   started_ = false;
   if (checkForErrors) {
     // Check for errors
-    std::lock_guard lock(error_mutex_);
+    std::exception_ptr err;
     for (size_t i = 0; i < threads_.size(); ++i) {
-      if (!tl_errors_[i].empty()) {
+      if (!err && !tl_errors_[i].empty()) {
         // Throw the first error that occurred
-        string error = make_string("Error in thread ", i, ": ", tl_errors_[i].front());
-        tl_errors_[i].pop();
```

Comment:
This was a bug - we only popped one error and a subsequent call to WaitForWork after unrelated work has been scheduled, would throw any remaining errors.